### ISSUE TITLE
Feature request: links in paragraphs

### ISSUE THREAD

**Issue by cwidhelm** [iainbrighton/PScribo#112]
Can you add support for a link tag?

Either in a paragraph or in a text block it would be nice to be able to inline a link $(Link -href "https://google.com" -target _blank -Label {Click Here}) so we can create more dynamic content linking to other resources.

**Comment by rebelinux** [comment]
I think this would be a great feature!

**Comment by Atreidae** [comment]
This would be handy in Tables as well as Paragraphs, As this module is used heavily in AsBuiltReport this would allow us to have a table of configuration items, and then the clickable link to jump to that item.

For example, Say an object is comprised of multiple smaller objects. The table can link to each of those smaller objects.

**Comment by iainbrighton** [comment]
I think adding external/web links to paragraphs would be a fairly simple implementation. 
```
{
    Paragraph {
        Text 'You can find the PScribo documentation examples'
        Link -Text 'Here' -Uri 'https://github.com/iainbrighton/PScribo/tree/dev/Examples' -Target New
    }
}
```
Would render this (Markdown doesn't support the target attribute)
> You can find the PScribo documentation examples [here](https://github.com/iainbrighton/PScribo/tree/dev/Examples)

Adding them to tables would be a lot more difficult and would need further discussions before any implementation.

**Comment by iainbrighton** [comment]
> This would be handy in Tables as well as Paragraphs, As this module is used heavily in AsBuiltReport this would allow us to have a table of configuration items, and then the clickable link to jump to that item.
> 
> For example, Say an object is comprised of multiple smaller objects. The table can link to each of those smaller objects.

@Atreidae you are talking about adding cross-references here - not external/web links? Adding support for this would be fairly straightforward for paragraphs, but a lot more involved for tables (just like with web links). I'll have a think about how it could be implemented.

**Comment by Atreidae** [comment]
Yes, that's correct, I'm looking for a way to implement similar functionality to tools like AudioCodes Ini file viewer.
With, Teams we have several objects that reference other objects (Dial Plans that reference multiple normalization rules, for example), and it's handy to be able to jump from one to another.

For example, see the aforementioned Ini viewer.
![image](https://github.com/iainbrighton/PScribo/assets/8736291/2b373ade-816a-4ee3-8a1b-e7e719f9d8a9)

Clicking on this "2" takes us straight to the **IP Group** section to view the relevant entry.
![image](https://github.com/iainbrighton/PScribo/assets/8736291/94e3c0c7-da02-4979-9bff-17d5af112c02)


As for the **link** subtype in the new paragraph method, that would still be very handy for references to external documentation.

![image](https://github.com/iainbrighton/PScribo/assets/8736291/7950da9d-6395-4499-8cdf-e32c6739f206)

**Comment by Atreidae** [comment]
Had a play around this afternoon. I'm slowly wrapping my head around how this works. 
This code is so much more beautiful than mine. Anyway.

I've hacked together a working implementation, just for text files for now, 
https://github.com/iainbrighton/PScribo/compare/dev...Atreidae:PScribo:dev
It takes a Link statement in a paragraph run and adds it via some changes to Out-TextParagraph

Hopefully, I'm sticking with the convention enough, if so, I keep developing and see if I can get it working with Word and HTML for linking to existing anchors.

**Comment by Atreidae** [comment]
@iainbrighton any thoughts before I keep going?

**Comment by jonmundle** [comment]
Are there any updates to this? This would be incredibly nice to have in either a paragraph or table.
